code/modules/mod/mod_activation.dm

```diff
@@ -59,6 +61,7 @@
 		span_notice("[src] [deploy ? "deploys" : "retracts"] its' parts with a mechanical hiss."),
 		span_hear("You hear a mechanical hiss."))
 	playsound(src, 'sound/mecha/mechmove03.ogg', 25, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	SEND_SIGNAL(src, COMSIG_MOD_PART_TOGGLED, user)
```

Comment:
same thing as before here, use only 1 signal type per event. Put these into a proc instead that gets called, e.g. on_mod_part_toggled

Reply:
Uh, what do you mean by this? Do the same thing I did with COMSIG_MOD_MODULE_CHANGED?